stsynphot/tests/test_parser.py

```diff
@@ -27,52 +31,242 @@
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_ebmvx_mwavg():
+    sp1 = spparser.parse_spec('ebmvx(0.3, mwavg)')
+    _single_functioncall(
+        sp1, ExtinctionCurve, Empirical1D, 'ebmvx(0.3,mwavg)', ans_z=None)
+
+    sp2 = spectrum.ebmvx('mwavg', 0.3)
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_rn_calspec_box():
+    sp1 = spparser.parse_spec(
+        'rn(crcalspec$gd71_mod_005.fits, box(5000, 10), 17, vegamag)')
+    _single_functioncall(
+        sp1, SourceSpectrum, None,
+        'rn(crcalspec$gd71_mod_005.fits,box(5000.0,10.0),17.0,vegamag)')
+
+    gd71 = SourceSpectrum.from_file(os.path.join(
+        os.environ['PYSYN_CDBS'], 'calspec', 'gd71_mod_005.fits'))
+    box = SpectralElement(Box1D, amplitude=1, x_0=5000 * u.AA, width=10 * u.AA)
+    sp2 = gd71.normalize(17 * units.VEGAMAG, band=box, vegaspec=spectrum.Vega)
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_rn_icat_k93():
+    sp1 = spparser.parse_spec(
+        'rn(icat(k93models, 5000, 0.5, 0), '
+        'cracscomp$acs_f814w_hrc_006_syn.fits, 17, obmag)')
+    _single_functioncall(
+        sp1, SourceSpectrum, None,
+        'rn(k93models(T_eff=5000,metallicity=0.5,log_g=0),'
+        'cracscomp$acs_f814w_hrc_006_syn.fits,17.0,obmag)')
+
+    k93 = catalog.grid_to_spec('k93models', 5000, 0.5, 0)
+    bp = SpectralElement.from_file(os.path.join(
+        os.environ['PYSYN_CDBS'], 'comp', 'acs', 'acs_f814w_hrc_006_syn.fits'))
+    sp2 = k93.normalize(17 * units.OBMAG, band=bp, area=conf.area)
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_rn_powerlaw():
+    sp1 = spparser.parse_spec('rn(pl(5000, 1, flam), band(v), 1, photlam)')
+    _single_functioncall(sp1, SourceSpectrum, None,
+                         'rn(pl(5000.0,1.0,flam),band(v),1.0,photlam)')
+
+    pl = SourceSpectrum(PowerLawFlux1D, amplitude=1 * units.FLAM,
+                        x_0=5000 * u.AA, alpha=-1)
+    bp = SpectralElement.from_filter('johnson_v')
+    sp2 = pl.normalize(1 * units.PHOTLAM, band=bp)
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_rn_unit_1_flam():
+    sp1 = spparser.parse_spec(
+        'rn(unit(1,flam), band(acs, wfc1, fr388n#3881.0), 10, abmag)')
+    _single_functioncall(
+        sp1, SourceSpectrum, None,
+        'rn(unit(1.0,flam),band(acs,wfc1,fr388n#3881.0),10.0,abmag)')
+
+    constsp = SourceSpectrum(ConstFlux1D, amplitude=1 * units.FLAM)
+    bp = spectrum.band('acs, wfc1, fr388n#3881.0')
+    sp2 = constsp.normalize(10 * u.ABmag, band=bp)
+    _compare_spectra(sp1, sp2)
+
+
+@pytest.mark.remote_data
+def test_remote_rn_calspec_u():
+    sp1 = spparser.parse_spec(
+        'rn(crcalspec$bd_75d325_stis_002.fits, band(u), 9.5, vegamag) * '
+        'band(fos, blue, 4.3, g160l)')
+    # NOTE: No expr for this combo.
+    _single_functioncall(sp1, SourceSpectrum, None, '')
```

Comment:
@ibusko , as discussed in spacetelescope/synphot_refactor#237, this is the only string among those being tested that does not have a "name."